wp-content/plugins/core/src/Blocks/Block_View_Factory.php

```diff
@@ -0,0 +1,111 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Project\Blocks;
+
+use DI\FactoryInterface;
+use WP_Block;
+
+/**
+ * @TODO make an interface for this.
+ * @TODO move to tribe-libs.
+ */
+class Block_View_Factory {
+
+	public string $block_path;
+
+	private FactoryInterface $container;
+
+	public function __construct( FactoryInterface $container, string $block_path ) {
+		$this->container  = $container;
+		$this->block_path = $block_path;
+	}
+
+	/**
+	 * Build the block model, controller and render the block view.
+	 *
+	 * @filter tribe/project/block/render
+	 *
+	 * @param array          $block
+	 * @param string         $content
+	 * @param bool           $is_preview
+	 * @param int            $post_id
+	 * @param \WP_Block|null $WP_block
+	 * @param array          $context
+	 *
+	 * @throws \DI\DependencyException
+	 * @throws \DI\NotFoundException
+	 */
+	public function render_template( array $block, string $content, bool $is_preview, int $post_id, ?WP_Block $WP_block, array $context = [] ): void {
```

Comment:
This is where the model and controller are instantiated because it's the only time the data we need is available.

We can fine tune control of what gets passed to the view as well which was a thing that is badly missing from `get_template_part()`, but standard blocks really just need the controller.